File: website/app/models/Country.scala
package models

import play.api.libs.json.{Json, JsValue, Writes}

case class Country(id: Long,
                   name: String)

object Country {
  implicit val writes = new Writes[Country] {
    def writes(country: Country): JsValue = {
      Json.obj(
        "id" -> country.id,
        "name" -> country.name
      )
    }
  }

  val Global = Country(0, "Global")
}

File: website/javascripts/models/country.js
CBR.Models.Country = P(CBR.Models.JsonSerializable, function (m) {
    m.options = {  // Defaults
    };

    m.getId = function () {
        return this.options.id;
    };

    m.getName = function () {
        return this.options.name;
    };
});

CBR.Models.Country.global = CBR.Models.Country({
    id: 0,
    name: "Global"
});
